libraries/AP_HAL_Linux/Storage_FRAM.cpp

```diff
@@ -242,20 +241,20 @@ int8_t Storage_FRAM::_register_read( uint16_t addr, uint8_t opcode )
     uint8_t tx[4] = {opcode, (uint8_t)((addr >> 8U)), (uint8_t)(addr & 0xFF), 0};
     uint8_t rx[4];
     
-    if(transaction(tx, rx, 4) == -1){
+    if(transaction(tx, rx) == -1){
         return -1;
     }
     return rx[3];
 }
 
-int8_t Storage_FRAM::transaction(uint8_t* tx, uint8_t* rx, uint16_t len){
-    _spi_sem = _spi->get_semaphore();
-    if (!_spi_sem->take(100)) {
+int8_t Storage_FRAM::transaction(uint8_t* tx, uint8_t* rx){
+
+    if (!_dev->get_semaphore()->take(100)) {
        // FRAM: Unable to get semaphore
         return -1;
     }
-    _spi->transaction(tx, rx, len);
-    _spi_sem->give();
+    _dev->transfer(tx, sizeof(*tx), rx, sizeof(*rx));
```

Comment:
I'm not sure if we actually need the local `transaction()` method. This is just abstracting the lock/unlock around the _dev->transfer() method. We could just live without it.